src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -1042,17 +1042,49 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
             ISignedPackageReader signedPackageReader,
             CancellationToken token)
         {
-            if (packageExtractionContext.SignedPackageVerifier != null)
+            if (packageExtractionContext.ClientPolicyContext != null)
             {
+                var clientPolicyContext = packageExtractionContext.ClientPolicyContext;
                 var repositorySignatureInfo = GetRepositorySignatureInfo(source);
 
-                var verifierSettings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(
+                var verificationSettings = RepositorySignatureInfoUtility.GetSignedPackageVerifierSettings(
                     repositorySignatureInfo,
-                    packageExtractionContext.SignedPackageVerifierSettings);
+                    clientPolicyContext.VerifierSettings);
 
-                var verifyResult = await packageExtractionContext.SignedPackageVerifier.VerifySignaturesAsync(
+                IPackageSignatureVerifier signedPackageVerifier = null;
+                if (packageExtractionContext.SignedPackageVerifier != null)
```

Comment:
Is the SignedPackageVerifier  property only used for tests? 
Can it be overriden somehow differently?

You wouldn't be able to use a factory method for the default providers in that case.

Reply:
Technically, every scenario should use the verification providers it needs. Since we don't have different scenarios for different restore code paths (and I can't think of why we should have) it seemed better to create the verifier in a central place (i.e. in this method). Some tests relied on being able to mock the verifier, therefore I made this property `internal` so that tests could continue to mock it, while real scenarios can avoid creating it in multiple places.